jaeger-core/src/main/java/io/jaegertracing/internal/TracingFactory.java

```diff
@@ -0,0 +1,77 @@
+package io.jaegertracing.internal;
+
+import io.jaegertracing.internal.clock.Clock;
+import io.jaegertracing.internal.metrics.Metrics;
+import io.jaegertracing.spi.BaggageRestrictionManager;
+import io.jaegertracing.spi.Reporter;
+import io.jaegertracing.spi.Sampler;
+import io.opentracing.ScopeManager;
+
+import java.util.List;
+import java.util.Map;
+
+public class TracingFactory {
+  public JaegerTracer createTracer(
+      String serviceName,
+      Reporter reporter,
+      Sampler sampler,
+      PropagationRegistry registry,
+      Clock clock,
+      Metrics metrics,
+      Map<String, Object> tags,
+      boolean zipkinSharedRpcSpan,
+      ScopeManager scopeManager,
+      BaggageRestrictionManager baggageRestrictionManager,
+      boolean expandExceptionLogs) {
+    return new JaegerTracer(
```

Comment:
per my comment on Configuration, I don't think this method needs to be in the factory. Here's what you can do instead, to reduce the surface area:
```java
class Configuration {
  public JaegerTracer.Builder getTracerBuilder() {
    - JaegerTracer.Builder builder = tracingFactory.createTracerBuilder(serviceName)
    + JaegerTracer.Builder builder = this.createTracerBuilder(serviceName)
  }

  protected createTracerBuilder() // overridable method, no more dependency on the factory
}

similar thing with JaegerTracer.Builder - just implement a protected method to create the tracer.
```

The only time where you need the abstract factory pattern is for creating SpanBuilder/Span/SpanContext, because those are done all the time from many places. But Tracer.Builder and Tracer are only created once, there's no need to complicate that with abstract factory pattern.

Reply:
I thought about doing that, but sort of wanted to do everything in factory or everything as subclass methods. This makes sense as a compromise.